### PR TITLE
Update the external PMIx component to match the PMIx internal integration

### PR DESCRIPTION
**Pull request by rhc54**
Solves observed differences between use of internal and external PMIx libraries

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1579/ for details.

File: opal/mca/pmix/external/pmix_ext.c [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * Copyright (c) 2014-2015 Intel, Inc.  All rights reserved.
  * Copyright (c) 2014-2015 Research Organization for Information Science
  *                         and Technology (RIST). All rights reserved.
- * Copyright (c) 2014-2015 Mellanox Technologies, Inc.
+ * Copyright (c) 2014      Mellanox Technologies, Inc.
```
Comment (hppritcha) [inline opal/mca/pmix/external/pmix_ext.c:6]: what happened here?

Comment (rhc54) [inline opal/mca/pmix/external/pmix_ext.c:6]: In order to sync the external glue code to the internal one, I did a simple copy of the internal to the external, and then updated the names to the new component. So the copyrights and everything else just flowed across.

**Comment by hppritcha** [comment]
@ggouaillardet please review as soon as possible - we'd like to get rc3 out asap.

**Comment by ggouaillardet** [comment]
i will be back at work on Friday only (this is Golden Week in Japan), and I might not be able to correctly review this before

**Comment by ggouaillardet** [comment]
@rhc54 you might need open-mpi/ompi@45f9a47d77f0ea1c562dd6f3bd59a24a25a23287 as well

**Comment by rhc54** [comment]
@ggouaillardet Okay, I added it

@hppritcha I'd recommend someone else review this to avoid the delay - or I can provide an MTT run based on it if you prefer.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1594/ for details.

**Comment by hppritcha** [comment]
@rhc54 let's go with a MTT run.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1597/ for details.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by rhc54** [comment]
``` shell
+-------------+-----------------+-------------+----------+------+------+----------+------+---------------------------------------------------------------------------+
| Phase       | Section         | MPI Version | Duration | Pass | Fail | Time out | Skip | Detailed report                                                           |
+-------------+-----------------+-------------+----------+------+------+----------+------+---------------------------------------------------------------------------+
| MPI Install | my installation | 2.0.0rc3    | 00:02    | 1    |      |          |      | MPI_Install-my_installation-my_installation-2.0.0rc3-my_installation.html |
| Test Build  | trivial         | 2.0.0rc3    | 00:00    | 1    |      |          |      | Test_Build-trivial-my_installation-2.0.0rc3-my_installation.html          |
| Test Build  | ibm             | 2.0.0rc3    | 00:45    | 1    |      |          |      | Test_Build-ibm-my_installation-2.0.0rc3-my_installation.html              |
| Test Build  | intel           | 2.0.0rc3    | 01:29    | 1    |      |          |      | Test_Build-intel-my_installation-2.0.0rc3-my_installation.html            |
| Test Build  | onesided        | 2.0.0rc3    | 00:05    | 1    |      |          |      | Test_Build-onesided-my_installation-2.0.0rc3-my_installation.html         |
| Test Build  | java            | 2.0.0rc3    | 00:03    | 1    |      |          |      | Test_Build-java-my_installation-2.0.0rc3-my_installation.html             |
| Test Build  | orte            | 2.0.0rc3    | 00:01    | 1    |      |          |      | Test_Build-orte-my_installation-2.0.0rc3-my_installation.html             |
| Test Run    | trivial         | 2.0.0rc3    | 00:04    | 6    |      |          |      | Test_Run-trivial-my_installation-2.0.0rc3-my_installation.html            |
| Test Run    | ibm             | 2.0.0rc3    | 08:13    | 376  | 1    |          | 3    | Test_Run-ibm-my_installation-2.0.0rc3-my_installation.html                |
| Test Run    | spawn           | 2.0.0rc3    | 00:07    | 7    |      |          |      | Test_Run-spawn-my_installation-2.0.0rc3-my_installation.html              |
| Test Run    | loopspawn       | 2.0.0rc3    | 10:00    | 1    |      |          |      | Test_Run-loopspawn-my_installation-2.0.0rc3-my_installation.html          |
| Test Run    | intel           | 2.0.0rc3    | 23:08    | 473  | 1    |          | 4    | Test_Run-intel-my_installation-2.0.0rc3-my_installation.html              |
| Test Run    | intel_skip      | 2.0.0rc3    | 11:31    | 431  |      |          | 47   | Test_Run-intel_skip-my_installation-2.0.0rc3-my_installation.html         |
| Test Run    | onesided        | 2.0.0rc3    | 00:25    | 32   |      |          |      | Test_Run-onesided-my_installation-2.0.0rc3-my_installation.html           |
| Test Run    | java            | 2.0.0rc3    | 00:01    | 1    |      |          |      | Test_Run-java-my_installation-2.0.0rc3-my_installation.html               |
| Test Run    | orte            | 2.0.0rc3    | 00:44    | 19   |      |          |      | Test_Run-orte-my_installation-2.0.0rc3-my_installation.html               |
+-------------+-----------------+-------------+----------+------+------+----------+------+---------------------------------------------------------------------------+


    Total Tests:    1355
    Total Failures: 2
    Total Passed:   1353
    Total Duration: 3398 secs. (56:38)
```

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1599/ for details.

**Comment by jsquyres** [comment]
@hppritcha Is this now good to go?

**Pull request merged**